Valentine Kopotchel/0/bread_bot.rb

```diff
@@ -0,0 +1,321 @@
+  '2016-10' => 'data/Prices_goods_10_2016.xlsx',
+  '2016-11' => 'data/Average prices_tov_11_2016.xlsx',
+  '2016-12' => 'data/prices_tov-12-2016.xlsx',
+  '2017-01' => 'data/price-01-2017.xlsx',
+  '2017-02' => 'data/Average_prices_02-2017.xlsx',
+  '2017-03' => 'data/Average_prices(serv)-03-2017.xlsx',
+  '2017-04' => 'data/Average_prices(serv)-04-2017.xlsx',
+  '2017-05' => 'data/Average prices(serv)-05-2017.xlsx',
+  '2017-06' => 'data/Average_prices(serv)-06-2017.xls',
+  '2017-07' => 'data/Average_prices(serv)-07-2017-2.xlsx',
+  '2017-08' => 'data/Average_prices(serv)-08-2017.xlsx',
+  '2017-09' => 'data/Average_prices(serv)-09-2017.xls',
+  '2017-10' => 'data/Average_prices(serv)-10-2017.xlsx',
+  '2017-11' => 'data/Average prices(serv)-07-2018.xlsx',
+  '2017-12' => 'data/Average prices(serv)-12-2017.xlsx',
+  '2018-01' => 'data/Average_prices(serv)-01-2018.xlsx',
+  '2018-02' => 'data/Average_prices(serv)-02-2018.xlsx',
+  '2018-03' => 'data/Average_prices(serv)-03-2018.xlsx',
+  '2018-04' => 'data/Average_prices(serv)-04-2018.xlsx',
+  '2018-05' => 'data/Average prices(serv)-05-2018.xlsx',
+  '2018-06' => 'data/Average_prices(serv)-06-2018.xlsx',
+  '2018-07' => 'data/Average prices(serv)-07-2018.xlsx',
+  '2018-08' => 'data/Average_prices(serv)-08-2018.xlsx',
+  '2018-09' => 'data/Average_prices(serv)-09-2018.xlsx',
+  '2018-10' => 'data/Average_prices(serv)-10-2018.xlsx'
+}.freeze
+# Doing some dirty work to make less code in BreadBot
+class BotHelper
+  PRICE_COLUMN_MINSK = 16
+  NAME_COLUMN = 0
+
+  def build_page(key)
+    path = HASH_OF_TABLES[key]
+    xlsx = Roo::Spreadsheet.open(path)
+    xlsx.sheet 0
+  end
+
+  def deflation?(date)
+    date[0, 4] < '2017'
+  end
+
+  def exists?(arg)
+    arg.to_s.length.positive?
+  end
+
+  def num?(arg)
+    arg.is_a? Numeric
+  end
+
+  def build_hash(key)
+    fin_hash = {}
+    condition = deflation? key
+    build_page(key).each do |row|
+      next unless exists?(row[NAME_COLUMN]) && num?(row[PRICE_COLUMN_MINSK])
+
+      val = row[PRICE_COLUMN_MINSK]
+      val /= 10_000.0 if condition
+      fin_hash[row[NAME_COLUMN]] = val
+    end
+    fin_hash
+  end
+
+  def make_normal(str)
+    str.swapcase.capitalize
+  end
+
+  def reg_build(str)
+    str = str.strip
+    r = str.upcase
+    temp = Regexp.new("(#{r}$|#{r}\s|#{r},)")
+    Regexp.and_union(/(^|[\P{L}])/, temp)
+  end
+
+  def split_key(str)
+    [str[0, 4], str[-2, 2]]
+  end
+
+  def minmax_assign(hash)
+    min = hash.keys.min
+    max = hash.keys.max
+    a = hash[min]
+    b = hash[max]
+    [min, max, a, b]
+  end
+
+  def write_minmax_info(hash)
+    min, max, a, b = minmax_assign(hash)
+    puts "Lowest was on #{(split_key a)[1]}/#{(split_key a)[0]} "\
+"at #{min.round(2)} BYN"
+    puts "Highest was on #{(split_key b)[1]}/#{(split_key b)[0]} "\
+"at #{max.round(2)} BYN"
+  end
+end
+# The main class
+class BreadBot
+  def initialize
+    @big_hash = {}
+    dog = BotHelper.new
```

Comment:
uncommunicative variable name
BotHelper should be a Module
You can include it and call methods directly